pkg/templates/nstemplatetiers/nstemplatetier_generator.go

```diff
@@ -63,20 +64,41 @@ func removeNoLongerBundledTiers(ctx context.Context, client runtimeclient.Client
 	for _, tier := range allTiers.Items {
 		if tier.Annotations[toolchainv1alpha1.BundledAnnotationKey] == constants.BundledWithHostOperatorAnnotationValue &&
 			!slices.Contains(bundledTierKeys, runtimeclient.ObjectKeyFromObject(&tier)) {
+			// delete the tier and all its templates
 			if err := client.Delete(ctx, &tier); err != nil {
 				allErrors = append(allErrors, err)
 			}
+
+			if err := deleteReferencedTierTemplates(ctx, client, &tier); err != nil {
+				allErrors = append(allErrors, err)
+			}
 		}
 	}
 
-	err := errors.Join(allErrors...)
+	err := stderrors.Join(allErrors...)
 	if err != nil {
 		err = fmt.Errorf("failed to delete some of the no-longer-bundled NSTemplateTiers: %w", err)
 	}
 
 	return err
 }
 
+func deleteReferencedTierTemplates(ctx context.Context, client runtimeclient.Client, tier *toolchainv1alpha1.NSTemplateTier) error {
+	for _, ref := range GetNSTemplateTierRefs(tier) {
+		template := &toolchainv1alpha1.TierTemplate{}
+		if err := client.Get(ctx, runtimeclient.ObjectKey{Name: ref, Namespace: tier.Namespace}, template); err != nil {
+			if kerrors.IsNotFound(err) {
+				continue
```

Comment:
should we add some logger/statement so we would know what happened.?

Reply:
I was thinking the same, but maybe logging what was deleted , not what was skipped, not found.

Reply:
done